scripts/core/editor3/helpers/highlights.js

```diff
@@ -394,6 +428,101 @@ export function getCharByOffset(editorState, selection, offset) {
     return block.getText()[newOffset];
 }
 
+/**
+ * @ngdoc method
+ * @name changeEditorSelection
+ * @param {Object} editorState
+ * @param {Integer} startOffset - the anchor offset relative to current start offset
+ * @param {Integer} endOffset - the focus offset relative to current end offset
+ * @param {Boolean} force - apply accept or force selection
+ * @return {Object} returns new state
+ * @description Change the current editor selection.
+ */
+export function changeEditorSelection(editorState, startOffset, endOffset, force) {
```

Comment:
Could you explain me how this works and why is it needed?

Reply:
This is an old function from suggestion reducer moved and improved here. I seen now that you added  expandDraftSelection and those two functions are doing the same thing. I can use expandDraftSelection but you have to add support for negative values for stretchLeft and stretchRight.

Reply:
Ok, I'll add support for negative values.

Reply:
ok

Reply:
Probably the name should be changed too.

Reply:
I'm thinking about `resizeSelection`, let me know if you have any suggestions

Reply:
It is ok

Reply:
If it is too complicate to do that change we can continue to use changeEditorSelection because this function already has that functionality.
We can easily change changeEditorSelection to don't update the editor state so we can use it where expandDraftSelection is used now.

Reply:
I thought it's not too complicated, seems like it's not that simple, but I'm almost done :)

Reply:
ok

Reply:
Done #2153

Reply:
The small refactoring required here can be done part of  #2153.
All code except the setting of the selection can be extracted as expandDraftSelection and then add a call to this function here.